src/optimizer.rs

```diff
@@ -284,7 +287,7 @@ mod optimization_tests_pattern_group {
 
         let fused = optimization.fusion(&filters);
 
-        assert!(fused.is_regex(), "Expected rule to be regex");
+        assert!(fused.is_regex() == false, "Expected rule to not be a regex");
```

Comment:
Now it checks non-regex case. We could also add another case if needed.

Reply:
Sounds like there are gonna be some changes here anyways so this should be fine for now